### PR TITLE
Coverity: Change many auto calls to const auto& to avoid unnecessary copies.

### PR DESCRIPTION
**Pull request by JohnAFernandez**
Technically these are all optimizations.  Some of them might even have the tiniest impact.

There are also a couple of very small refactors that also avoid unnecessary copies instead of usting auto &.

### CODE REVIEW

File: code/scripting/doc_parser.cpp [hunk 1]
```diff
@@ -202,8 +202,8 @@ class TypeVisitor : public BaseVisitor {
 
 	antlrcpp::Any visitMap_type(ArgumentListParser::Map_typeContext* context) override
 	{
-		const auto keyType = visit(context->type(0)).as<ade_type_info>();
-		const auto valueType = visit(context->type(1)).as<ade_type_info>();
+		auto keyType = visit(context->type(0)).as<ade_type_info>();
+		auto valueType = visit(context->type(1)).as<ade_type_info>();
```
Comment (Goober5000) [inline code/scripting/doc_parser.cpp:206]: Why remove the `const` here?

File: test/src/scripting/doc_parser.cpp [hunk 2]
```diff
@@ -242,10 +242,10 @@ TEST_F(ArgumentListParserTest, CommentOnArgWithFolliwingArg) {
 TEST_F(ArgumentListParserTest, MapType) {
 	ASSERT_TRUE(parser.parse("{ string => number ... } mapArg"));
 
-	const auto arglist = parser.getArgList();
+	const auto& arglist = parser.getArgList();
 	ASSERT_EQ(1, static_cast<int>(arglist.size()));
 
-	const auto mapType = arglist[0].type;
+	auto mapType = arglist[0].type;
```
Comment (Goober5000) [inline test/src/scripting/doc_parser.cpp:248]: Why remove the `const` here?

File: test/src/scripting/doc_parser.cpp [hunk 3]
```diff
@@ -262,10 +262,10 @@ TEST_F(ArgumentListParserTest, MapType) {
 TEST_F(ArgumentListParserTest, IteratorType) {
 	ASSERT_TRUE(parser.parse("iterator<string> iteratorArg"));
 
-	const auto arglist = parser.getArgList();
+	const auto& arglist = parser.getArgList();
 	ASSERT_EQ(1, static_cast<int>(arglist.size()));
 
-	const auto mapType = arglist[0].type;
+	auto mapType = arglist[0].type;
```
Comment (Goober5000) [inline test/src/scripting/doc_parser.cpp:268]: Why remove the `const` here?

**Review by Goober5000 — COMMENTED** [review]
A comment and some questions.

File: code/graphics/opengl/gropenglpostprocessing.cpp [hunk 4]
```diff
@@ -851,8 +851,8 @@ void opengl_post_shader_header(SCP_stringstream& sflags, shader_type shader_t, i
 		}
 	} else if (shader_t == SDR_TYPE_POST_PROCESS_LIGHTSHAFTS) {
 		char temp[64];
-		const auto ls_params = graphics::Post_processing_manager->getLightshaftParams();
-		sprintf(temp, "#define SAMPLE_NUM %d\n", ls_params.samplenum);
+		const auto& ls_params = graphics::Post_processing_manager->getLightshaftParams();
+		snprintf(temp, 64, "#define SAMPLE_NUM %d\n", ls_params.samplenum);
```
[the remark this answers is not in the document]
Comment (JohnAFernandez) [reply code/graphics/opengl/gropenglpostprocessing.cpp:855]: I don't like putting a lot of unrelated changes together.

File: code/scripting/doc_parser.cpp (hunk 1 above)
Comment (JohnAFernandez) [reply code/scripting/doc_parser.cpp:206]: Uh ... let me reverse those and the other accidental const changes.

File: code/graphics/opengl/gropenglpostprocessing.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (Goober5000) [reply code/graphics/opengl/gropenglpostprocessing.cpp:855]: Ok

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**